### ISSUE TITLE
cmake -DBUILD_SHARED_LIBS=ON requires -DBUILD_TZ_LIB=ON

### ISSUE THREAD

**Issue by mechanyx** [HowardHinnant/date#566]
TLDR; Previously, setting -DBUILD_SHARED_LIBS=ON would build libtz.so but now it does nothing by itself. You must also include -DBUILD_TZ_LIB=ON

Up until October 11th's commit 48f1455cd255df2fe73b5f293b47203ad0988199 "CMake refactor for easier subdirectory inclusion", you could build the shared library libtz.so by doing something like:

```
git clone
mkdir build
cd build
cmake /path/to/checkout -DBUILD_SHARED_LIBS=ON
make
```

But since that commit, the above does nothing. You have to change the cmake command to.

`cmake /path/to/checkout -DBUILD_SHARED_LIBS=ON -DBUILD_TZ_LIB=ON`

`BUILD_SHARED_LIBS` still does something because if you do just `-DBUILD_TZ_LIB=ON`, you get the .a instead of the .so but I think the old behavior is what most people expect.

Tangentially, while attempting to debug this, I saw that the cmake build instructions in the front page readme do not work.

**Comment by mellery451** [comment]
[BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) is a general CMake option that just determines the type of library that `add_library` will produce (if not explicitly specified). Prior to the commit you reference, the `tz`library was always being built, but it is really optional extensions and not considered part of the core (header-only) date library, thus we added the option that you've already discovered. This is the correct way to get `tz` currently, although is @HowardHinnant feels that `tz` should be promoted back to "always build" status, then this option can either be removed or defaulted to ON.
